File: LeetCode-In-Go/internal/medium/LeetCode_102.go
package medium

func levelOrder(root *TreeNode) [][]int {
	res := [][]int{}
	if root == nil {
		return res
	}

	stack := []*TreeNode{}
	stack = append(stack, root)

	for len(stack) > 0 {
		res1 := []int{}
		for _, node := range stack {
			res1 = append(res1, node.Val)
			stack = stack[1:]
			if node.Left != nil {
				stack = append(stack, node.Left)
			}
			if node.Right != nil {
				stack = append(stack, node.Right)
			}
		}
		res = append(res, res1)
	}
	return res
}

File: LeetCode-In-Java/src/BreadthFirstSearch/LeetCode_102.java
package BreadthFirstSearch;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

/**
 * TODO
 *
 * @author <a href="mailto:ge.sf.chn@gmail.com">shaofeng</a>
 * @see
 * @since 2019/5/15
 */
public class LeetCode_102 {

    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        List<List<Integer>> lists = new ArrayList<>();
        int size;
        while (root != null && (size = queue.size()) > 0) {
            List<Integer> list = new ArrayList<>();
            for (int i = 0; i < size; i++) {
                TreeNode cur = queue.poll();
                if (cur.left != null) {
                    queue.add(cur.left);
                }
                if (cur.right != null) {
                    queue.add(cur.right);
                }
                list.add(cur.val);
            }
            lists.add(list);
        }
        return lists;
    }
}
